ethereum/datastructures/src/main/java/tech/pegasys/artemis/datastructures/util/MerkleTree.java

```diff
@@ -22,40 +22,34 @@
 import org.apache.tuweni.crypto.Hash;
 import tech.pegasys.artemis.util.SSZTypes.SSZVector;
 
-public class MerkleTree<T> {
-  private final List<List<Bytes32>> tree;
-  private final List<Bytes32> zeroHashes;
-  private final int height;
-  private boolean dirty = true;
+abstract class MerkleTree {
+  protected final List<List<Bytes32>> tree;
+  protected final List<Bytes32> zeroHashes;
+  protected final int treeDepth;
 
-  public MerkleTree(int height) {
-    assert (height > 1);
-    this.height = height;
-    tree = new ArrayList<List<Bytes32>>();
-    for (int i = 0; i <= height; i++) {
-      tree.add(new ArrayList<Bytes32>());
+  protected MerkleTree(int treeDepth) {
+    this.treeDepth = treeDepth;
+    assert (treeDepth > 1);
```

Comment:
nit: usually worth having the assertions first and we probably should be consistent in using `checkArgument` rather than the `assert` keyword.

Reply:
done.